### ISSUE TITLE
License Scan/Remediation Questions

### ISSUE THREAD

**Issue by austinlparker** [open-telemetry/opentelemetry-go-instrumentation#697]
Hello maintainers! We have received a report from the CNCF license audit that the following paths contain licenses that do not match the project license. We will need to justify these dependencies for an exception from the CNCF Board. You can find the full report here: https://lfscanning.org/reports/cncf-2/open-telemetry-2024-02-09-e8fcbc12-0a27-470a-9f2d-c5f680322e13.html

- libbpf (LGPL 2.1)
- github.com/hashicorp/go-version (MPL 2.0)

I couldn't find any reference to libbpf in the repo, so I'm not sure what's up with that one, but it's the highest priority in terms of justifcation/removal per the report.

**Comment by damemi** [comment]
Hi @austinlparker!

The libbpf files were included for easier linkage during compilation of our C files. At that time, we found that it is dual-licensed under BSD (https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/4#issuecomment-1296849147) which is explicitly allowed according to https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/4#issuecomment-1302665304. Though, it sounds like we have alternatives to including those files. @edeNFed or @MrAlias did we ever change how we use them?

I don't think we actually have any files from hashicorp/go-version, it's just a dependency and we just included the license in our LICENSES folder for that. But we could probably refactor to use the native go/version package now that it's available anyway (https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/652)
